cmd/wallet-js-sdk/test/specs/httpsig/digest.spec.js

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+import { expect } from "chai";
+import getDigest from "../../../src/httpsig/digest";
+
+const expectThrowsAsync = async (method, errorMessage) => {
+  let error = null;
+  try {
+    await method();
+  } catch (err) {
+    error = err;
+  }
+  expect(error).to.be.an("Error");
+  if (errorMessage) {
+    expect(error.message).to.equal(errorMessage);
+  }
+};
+
+describe("HTTP Signature Module - Generating Digest", function () {
+  const mockData = {
+    access_token: [
+      {
+        access: [
+          {
+            type: "trustbloc.dev/types/gnap/userdata-access",
+            actions: ["read", "write", "update", "delete"],
+            locations: ["edv.provider.com/api"],
+            "subject-keys": ["sub"],
+          },
+          {
+            type: "trustbloc.dev/types/gnap/kms-access",
+            actions: ["sign", "verify", "encrypt", "decrypt"],
+            locations: ["kms.provider.com/api/user"],
+            "subject-keys": ["sub"],
+          },
+        ],
+        label: "access-1",
+      },
+      {
+        access: [
+          {
+            type: "trustbloc.dev/types/gnap/legacy-api",
+            actions: ["read", "append"],
+            locations: ["log.legacy.com/user"],
+          },
+        ],
+        label: "log-access",
+        flags: ["bearer"],
+      },
+    ],
+    client: {
+      key: {
+        proof: "httpsig",
+        jwk: {
+          kty: "EC",
+          kid: "key1",
+          crv: "P-256",
+          alg: "ES256",
+          x: "TDiE9hqocEOewpSgI5r_yhoOy1EuetUpWcKvlrC9Ix4",
+          y: "pC76IX_6YRQ5CaxmcmcixTA-dWjHpuKoYICir7VYQ_A",
```

Comment:
make these variables to allow different key types ( can be done in a future enhancement)

Reply:
Updated